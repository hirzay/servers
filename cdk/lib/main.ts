import { App } from "aws-cdk-lib";
import "source-map-support/register";
import { DomainStack } from "./domain/domain-stack";

const app = new App();
const props = {
  availabilityZone: "us-west-2a",
  env: {
    account: "692594597524",
    region: "us-west-2",
  },
};

new DomainStack(app, "DomainStack", props);
