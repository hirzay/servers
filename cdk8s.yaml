language: typescript
app: deno run --allow-write=dist src/main.ts
imports:
  - k8s
