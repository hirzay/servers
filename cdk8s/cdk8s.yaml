language: typescript
app: false
imports:
  - k8s
