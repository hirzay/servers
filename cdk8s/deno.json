{
  "tasks": {
    "build": "deno run --allow-write=dist --allow-read=config src/chart.ts",
    "up": "kubectl apply -f dist",
    "diff": "KUBECTL_EXTERNAL_DIFF=delta DELTA_PAGER=bat BAT_PAGER=/home/linuxbrew/.linuxbrew/bin/less kubectl diff -f dist/turing.k8s.yaml"
  }
}
