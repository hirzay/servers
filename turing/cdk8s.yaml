language: typescript
app: deno run --allow-write=dist/main.ts --allow-read=config ./src/main.ts
imports:
  - k8s
