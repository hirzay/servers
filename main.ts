import { App, Chart } from "npm:cdk8s";
import { Construct } from "npm:constructs";
import { KubeDeployment } from "./imports/k8s.ts";

export class MyChart extends Chart {
  constructor(scope: Construct, ns: string, appLabel: string) {
    super(scope, ns);

    new KubeDeployment(this, "my-deployment", {
      spec: {
        replicas: 3,
        selector: { matchLabels: { app: appLabel } },
        template: {
          metadata: { labels: { app: appLabel } },
          spec: {
            containers: [
              {
                name: "app-container",
                image: "nginx:1.19.10",
                ports: [{ containerPort: 80 }],
              },
            ],
          },
        },
      },
    });
  }
}

const app = new App();
new MyChart(app, "getting-started", "my-app");

app.synth();
