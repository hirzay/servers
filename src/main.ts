import { App, Chart } from "cdk8s";
import { createInvidiousDeployment } from "./invidious.ts";
import { createTedditDeployment } from "./teddit.ts";

const app = new App();
const chart = new Chart(app, "jerred");

createInvidiousDeployment(chart);
createTedditDeployment(chart);

app.synth();
