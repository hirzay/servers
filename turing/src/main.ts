import { App, Chart } from "npm:cdk8s";
import { createInvidiousDeployment } from "./services/frontends/invidious.ts";
import { createTedditDeployment } from "./services/frontends/teddit.ts";
import { createPlexDeployment } from "./services/media/plex.ts";
import { createRadarrDeployment } from "./services/torrents/radarr.ts";
import { createBazarrDeployment } from "./services/torrents/bazarr.ts";
import { createHomeAssistantDeployment } from "./services/homeassistant.ts";
import { createJackettDeployment } from "./services/torrents/jackett.ts";
import { createNitterDeployment } from "./services/frontends/nitter.ts";
import { createOverseerrDeployment } from "./services/torrents/overseerr.ts";
import { createQBitTorrentDeployment } from "./services/torrents/qbittorrent.ts";
import { createSonarrDeployment } from "./services/torrents/sonarr.ts";
import { createSyncthingDeployment } from "./services/syncthing.ts";
import { createTautulliDeployment } from "./services/media/tautulli.ts";
import { createGolinkDeployment } from "./services/golink.ts";
import { createBitmagnetDeployment } from "./services/torrents/bitmagnet.ts";

const app = new App();
const chart = new Chart(app, "turing", { namespace: "turing" });

createInvidiousDeployment(chart);
createTedditDeployment(chart);
createPlexDeployment(chart);
createRadarrDeployment(chart);
createBazarrDeployment(chart);
createHomeAssistantDeployment(chart);
createJackettDeployment(chart);
createNitterDeployment(chart);
createOverseerrDeployment(chart);
createQBitTorrentDeployment(chart);
createSonarrDeployment(chart);
createSyncthingDeployment(chart);
createTautulliDeployment(chart);
createGolinkDeployment(chart);
createBitmagnetDeployment(chart);

app.synth();
