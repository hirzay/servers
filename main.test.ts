import { MyChart } from "./main.ts";
import { Testing } from "npm:cdk8s";
import { assertSnapshot } from "https://deno.land/std@0.205.0/testing/snapshot.ts";

Deno.test("placeholder", async (t) => {
  const app = Testing.app();
  const chart = new MyChart(app, "test-chart", "label");
  const results = Testing.synth(chart);
  await assertSnapshot(t, results);
});
