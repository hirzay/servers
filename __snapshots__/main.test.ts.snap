export const snapshot = {};

snapshot[`placeholder 1`] = `
[
  {
    apiVersion: "apps/v1",
    kind: "Deployment",
    metadata: {
      name: "test-chart-my-deployment-c8d35e42",
    },
    spec: {
      replicas: 3,
      selector: {
        matchLabels: {
          app: "label",
        },
      },
      template: {
        metadata: {
          labels: {
            app: "label",
          },
        },
        spec: {
          containers: [
            {
              image: "nginx:1.19.10",
              name: "app-container",
              ports: [
                {
                  containerPort: 80,
                },
              ],
            },
          ],
        },
      },
    },
  },
]
`;
